sdk/core/core/src/az_span.c

```diff
@@ -231,15 +227,16 @@ AZ_NODISCARD az_result az_span_append(az_span destination, az_span source, az_sp
 {
   AZ_PRECONDITION_NOT_NULL(out_span);
 
-  int32_t const current_size = az_span_length(destination);
-  az_span remainder = az_span_slice(destination, current_size, -1);
-  AZ_RETURN_IF_FAILED(az_span_copy(remainder, source, &remainder));
-
-  *out_span = az_span_init(
-      az_span_ptr(destination),
-      current_size + az_span_length(source),
-      az_span_capacity(destination));
-
+  int32_t const dest_length = az_span_length(destination);
+  int32_t const dest_capacity = az_span_capacity(destination);
+  int32_t const src_length = az_span_length(source);
+  if ((dest_capacity - dest_length) < src_length)
+  {
+    return AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY;
+  }
+  uint8_t* ptr = az_span_ptr(destination);
+  memmove((void*)(&ptr[dest_length]), (void const*)az_span_ptr(source), (size_t)src_length);
+  *out_span = az_span_init(ptr, dest_length + src_length, dest_capacity);
```

Comment:
Can we call `az_span_copy` within `az_span_append` and avoid the duplication here or is the slice and call to copy too expensive?

Reply:
We should add a comment here that `dest_length + src_length` is guaranteed to not integer overflow because of the capacity check above.

Reply:
Sure, I love comments.

Reply:
instead of adding a comment, I have updated the logic a little bit to add an extra variable that holds src_len + dest_len.  Then use that one to validate it is still less than capacity and also for init span.. So we don't need to calculate remaining for the validation